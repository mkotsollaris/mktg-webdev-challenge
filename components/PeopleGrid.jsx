import { useContext } from "react"
import AppContext from "../context/AppContext"
import PersonCard from "./PersonCard";
import style from './style.module.css'

const PeopleGrid = () => {
    const { people } = useContext(AppContext);
    const element = people?.map(person => <PersonCard url={person?.avatar?.url} name={person.name} title={person.title} department={person.department.name} />)
    return <div className={style.people}>
        {element}
    </div>
}

export default PeopleGrid;