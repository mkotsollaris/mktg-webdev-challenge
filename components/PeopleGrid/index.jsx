import { useContext } from 'react'
import PersonCard from './PersonCard'
import style from './style.module.css'
import AppContext from '../../context/AppContext'

const PeopleGrid = () => {
  const { people } = useContext(AppContext)

  const People = (
    <div className={style.people}>
      {people?.map((person) => (
        <PersonCard
          key={person.id}
          url={person?.avatar?.url}
          name={person?.name}
          title={person?.title}
          department={person?.department?.name}
        />
      ))}
    </div>
  )

  const NoResultsFound = (
    <div className={style['no-results-found']}>No results found.</div>
  )
  return people?.length > 0 ? People : NoResultsFound
}

export default PeopleGrid
