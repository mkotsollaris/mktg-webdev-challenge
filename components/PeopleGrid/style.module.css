.people {
  display: grid;
  align-content: center;
  justify-content: center;
  grid-template-columns: 28% 28% 28%;
  grid-column-gap: 1rem;
  grid-row-gap: 1rem;
  padding: 2rem;

@media (max-width: 800px) {
    composes: people;
    grid-template-columns: 40% 40%
}

@media (max-width: 450px) {
    composes: people;
    grid-template-columns: 100%
}
}

.no-results-found {
  text-align: center;
  color: darkslategray;
  font-size: small;
}
