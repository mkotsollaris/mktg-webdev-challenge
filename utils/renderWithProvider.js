import { render } from '@testing-library/react'
import AppContext from '../context/AppContext'

function renderWithProvider(props, Component) {
  return render(
    <AppContext.Provider value={props}>
      <Component />
    </AppContext.Provider>
  )
}

export default renderWithProvider
