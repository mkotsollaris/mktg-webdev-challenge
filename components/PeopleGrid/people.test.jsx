import PeopleGrid from '.'
import renderWithProvider from '../../utils/renderWithProvider'

it('renders', () => {
  const { asFragment } = renderWithProvider({}, PeopleGrid)
  expect(asFragment()).toMatchSnapshot()
})
