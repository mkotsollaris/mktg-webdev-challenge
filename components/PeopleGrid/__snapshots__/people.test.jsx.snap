// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    No results found.
  </div>
</DocumentFragment>
`;
